packages/gatsby/src/utils/webpack.config.js

```diff
@@ -66,17 +66,40 @@ module.exports = async (
     return Object.assign(envObject, gatsbyVarObject)
   }
 
+  function getHmrPath () {
+    let hmrBasePath = `${
+      program.ssl ? `https` : `http`
+    }://${
+      program.host
+    }:${webpackPort}/`
+
+    const hmrSuffix = `__webpack_hmr&reload=true&overlay=false`
+
+    if (process.env.GATSBY_WEBPACK_PUBLICPATH) {
+      const pubPath = process.env.GATSBY_WEBPACK_PUBLICPATH
+      if (pubPath.substr(-1) === `/`) {
+        hmrBasePath = pubPath
+      } else {
+        hmrBasePath = `${pubPath}/`
+      }
+    }
+
+    return hmrBasePath + hmrSuffix
+  }
+
   debug(`Loading webpack config for stage "${stage}"`)
   function getOutput() {
     switch (stage) {
       case `develop`:
         return {
           path: directory,
           filename: `[name].js`,
-          publicPath: `http://${program.host}:${webpackPort}/`,
```

Comment:
This particular area made me nervous when I was resolving conflicts.  It seems like, in `v2`, ssl isn't supported.  Not sure if that was intentional